java-client/src/main/java/org/opensearch/client/opensearch/_types/query_dsl/QueryVariant.java

```diff
@@ -39,8 +39,12 @@ public interface QueryVariant {
 
     Query.Kind _queryKind();
 
+    @Deprecated
```

Comment:
Should we implement this one in terms of the other so we don't duplicate code? The JIT will optimize it away anyway.

Reply:
I've seen it being used elsewhere in code from this repository. The intention was to provide reference for internal code writers.